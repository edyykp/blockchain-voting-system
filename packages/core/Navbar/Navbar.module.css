.container {
  display: flex;
  justify-content: space-between;
  height: 80px;
  z-index: 1;
  width: 100%;
  padding: 0 24px;
  max-width: 1100px;
}

.logo {
  color: var(--primary_darker);
  justify-self: flex-start;
  cursor: pointer;
  font-size: 1.5rem;
  display: flex;
  align-items: center;
  margin-left: 24px;
  font-weight: bold;
  text-decoration: none;
}

.mobileMenuIcon {
  display: none;
}

.menu {
  display: flex;
  align-items: center;
  list-style: none;
  text-align: center;
  margin-right: -22px;
}

.menuItem {
  height: 80px;
}

.menuUrls {
  color: var(--secondary);
  display: flex;
  align-items: center;
  text-decoration: none;
  padding: 0 1rem;
  height: 100%;
  cursor: pointer;
}

.menuUrls:active {
  border-bottom: 3px solid var(--primary);
}

.cta {
  border-radius: 50px;
  background: var(--primary);
  white-space: nowrap;
  padding: 10px 22px;
  color: var(--secondary);
  font-size: 16px;
  outline: none;
  border: none;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
  text-decoration: none;
}
.cta:hover {
  transition: all 0.2s ease-in-out;
  background: var(--secondary);
  color: var(--primary_darker);
}

.ctaContainer {
  display: flex;
  align-items: center;
}

.faBars {
  color: var(--primary_darker);
}

@media screen and (max-width: 768px) {
  .mobileMenuIcon {
    display: block;
    position: absolute;
    top: 0;
    right: 0;
    transform: translate(-100%, 60%);
    font-size: 1.8rem;
    cursor: pointer;
  }

  .menu {
    display: none;
    color: var(--secondary);
  }

  .ctaContainer {
    display: none;
  }
}
