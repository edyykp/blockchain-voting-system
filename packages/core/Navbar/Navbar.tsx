import Link from 'next/link';
import { FaBars } from 'react-icons/fa';
import { Link as LinkS } from 'react-scroll';

import { useMenu, useSiteProperties } from '@packages/config';

import styles from './Navbar.module.css';

export const Navbar = () => {
  const menuItems = useMenu();
  const valueOf = useSiteProperties();

  const text = {
    logo: valueOf('site_title'),
    cta_title: valueOf('navbar_cta_title'),
    cta_link: valueOf('navbar_cta_link'),
  };

  return (
    <nav data-testid="navigation-bar">
      <div className={styles.container} data-testid="navigation-bar-logo">
        <Link href="/">
          <a className={styles.logo}>{text.logo}</a>
        </Link>
        <div className={styles.mobileMenuIcon} data-testid="mobile-menu-icon">
          <FaBars className={styles.faBars} />
        </div>
        <ul className={styles.menu} data-testid="navigation-bar-menu">
          {menuItems.map((item, key) => (
            <li className={styles.menuItem} key={key}>
              <LinkS to={item.url} className={styles.menuUrls}>
                {item.title}
              </LinkS>
            </li>
          ))}
        </ul>
        <div className={styles.ctaContainer} data-testid="navigation-bar-cta">
          <LinkS to={text.cta_link || ''} className={styles.cta}>
            {text.cta_title}
          </LinkS>
        </div>
      </div>
    </nav>
  );
};
