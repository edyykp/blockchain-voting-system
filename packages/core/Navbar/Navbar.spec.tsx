import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Navbar } from './Navbar';

const setup = () => {
  return {
    ...render(<Navbar />),
  };
};

describe('Navbar component', () => {
  it('renders', () => {
    const { container, getByTestId } = setup();

    expect(getByTestId('navigation-bar-desktop')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders the logo', () => {
    const { getByTestId } = setup();

    expect(getByTestId('navigation-bar-logo')).toBeInTheDocument();
  });

  it('renders the menu', () => {
    const { getByTestId } = setup();

    expect(getByTestId('navigation-bar-menu')).toBeInTheDocument();
  });

  it('renders the mobile menu icon', () => {
    const { getByTestId } = setup();

    expect(getByTestId('mobile-menu-icon')).toBeInTheDocument();
  });

  it('renders the cta', () => {
    const { getByTestId } = setup();

    expect(getByTestId('navigation-bar-cta')).toBeInTheDocument();
  });
});
