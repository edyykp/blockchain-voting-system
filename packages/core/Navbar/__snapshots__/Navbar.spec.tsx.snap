// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders 1`] = `
<div>
  <nav
    data-testid="navigation-bar-desktop"
  >
    <div
      class="container"
      data-testid="navigation-bar-logo"
    >
      <a
        class="logo"
        href="/"
      >
        BVS
      </a>
      <div
        class="mobileMenuIcon"
        data-testid="mobile-menu-icon"
      >
        <svg
          class="faBars"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          />
        </svg>
      </div>
      <ul
        class="menu"
        data-testid="navigation-bar-menu"
      >
        <li
          class="menuItem"
        >
          <a
            class="menuUrls"
          >
            About
          </a>
        </li>
        <li
          class="menuItem"
        >
          <a
            class="menuUrls"
          >
            Data Protection
          </a>
        </li>
      </ul>
      <div
        class="ctaContainer"
        data-testid="navigation-bar-cta"
      >
        <a
          class="cta"
        >
          Sign in
        </a>
      </div>
    </div>
  </nav>
</div>
`;
